src/motion/features/layout/Animate.tsx

```diff
@@ -281,19 +281,21 @@ class Animate extends React.Component<AnimateProps> {
             }
         })
 
-        // Force a render to ensure there's no flash of uncorrected bounding box.
-        visualElement.syncRender()
-
         /**
          * If this visualElement isn't present (ie it's been removed from the tree by the user but
          * kept in by the tree by AnimatePresence) then call safeToRemove when all axis animations
          * have successfully finished.
          */
-        return Promise.all(animations).then(() => {
-            this.isAnimatingTree = false
-            onComplete && onComplete()
-            visualElement.notifyLayoutAnimationComplete()
-        })
+        if (animations.filter(Boolean).length) {
+            // Force a render to ensure there's no flash of uncorrected bounding box.
+            visualElement.syncRender()
+
+            return Promise.all(animations).then(() => {
+                this.isAnimatingTree = false
```

Comment:
If `animate` has been called and `this.isAnimatingTree` set to `true`, when will it get set to `false` if no animations are created?

Reply:
I tried searching the codebase but I don't see it being used anywhere:

<img width="309" alt="image" src="https://user-images.githubusercontent.com/159500/121653046-93e68880-ca9c-11eb-9e16-93229114fc82.png">

Reply:
`shouldStackAnimate` is a Framer-internal thing so it isn't provided by anything in Motion